api/src/main/java/io/opencensus/metrics/Gauge.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2018, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.metrics;
+
+import io.opencensus.common.Clock;
+
+// TODO(mayurkale) : Add comments
+
+public interface Gauge {
```

Comment:
I would remove this completely. And make DataPoint not having a public function to getMetrics (this is an implementation detail).